### DISCUSSION TITLE
Is standard Lua supported instead of LuaJIT?

### DISCUSSION THREAD

**Discussion by oz-rw** [fluent/fluent-bit#8731] [Q&A]
After searching for `include "luajit.h"` in the codebase, it seems to me that the LuaJIT C API is NOT actually used. 
Under the directories `src/` and `plugins/` only the standard `"lua.h"` `"lauxlib.h"` `"lualib.h` from Lua are included.

Therefore it should be possible to use standard lua in its place? I modified the CMake configuration to accept a `LUA_DIR` with the Lua 5.4 source code instead of a `LUAJIT_DIR` and the lua filter seems to work fine.

Therefore I just wanted a confirmation that there isn't actual use of LuaJIT?

Because then I could open a PR for adding Lua5.4 support in the build.

**Comment by patrick-stephens** [comment]
I would just open a PR, ideally we can test it there but I suspect adding more LUA tests would be appreciated too.

**Reply by oz-rw** [reply]
Thanks for the quick reply.
I was planning to do that, but my changes would require about 3 lines of modification under the vendored `lib/luajit-cmake` "subproject". I suppose I should avoid that and try to fix/modify using the `cmake/luajit.cmake` wrapper?

**Reply by patrick-stephens** [reply]
Vendored changes are generally to be avoided - they should be via the upstream project - however it may demonstrate it works for then subsequent refactoring.
